py_swf/clients/workflow.py

```diff
@@ -65,3 +70,341 @@ def terminate_workflow(self, workflow_id, reason):
+            oldest_start_date=oldest_start_date,
+            latest_start_date=latest_start_date,
+        )
+        if not start_time_filter_dict:
+            return self.boto_client.count_open_workflow_executions(
+                domain=self.workflow_client_config.domain,
+                **workflow_filter_dict
+            )
+
+        return self.boto_client.count_open_workflow_executions(
+            domain=self.workflow_client_config.domain,
+            startTimeFilter=start_time_filter_dict['startTimeFilter'],
+            **workflow_filter_dict
+        )
+
+    def _count_closed_workflow_executions(
+            self,
+            oldest_start_date=None,
+            latest_start_date=None,
+            oldest_close_date=None,
+            latest_close_date=None,
+            workflow_filter_dict=None
+    ):
+        """
+        Count the number of closed workflows for a domain. You can pass in filtering criteria as a dict
+        The results are best effort and may not exactly reflect recent updates and changes.
+        http://docs.aws.amazon.com/amazonswf/latest/apireference/API_CountClosedWorkflowExecutions.html#SWF-CountClosedWorkflowExecutions-request-executionFilter
+        :param oldest_start_date: datetime
+        :param latest_start_date: datetime
+        :param oldest_close_date: datetime
+        :param latest_close_date: datetime
+        :param workflow_filter_dict: dict. filter type as key and filter criteria as value. filter criteria is a dictionary
+        :return: total number of closed workflows that meet the filter criteria
+        """
+        if not workflow_filter_dict:
+            workflow_filter_dict = {}
+
+        time_filter_dict = _build_time_filter_dict(
+            oldest_start_date=oldest_start_date,
+            latest_start_date=latest_start_date,
+            oldest_close_date=oldest_close_date,
+            latest_close_date=latest_close_date,
+        )
+        if not time_filter_dict:
+            return self.boto_client.count_closed_workflow_executions(
+                domain=self.workflow_client_config.domain,
+                **workflow_filter_dict
+            )
+
+        for key in time_filter_dict:
+            time_filter_type = key
+        if time_filter_type == 'startTimeFilter':
+            return self.boto_client.count_closed_workflow_executions(
+                domain=self.workflow_client_config.domain,
+                startTimeFilter=time_filter_dict[time_filter_type],
+                **workflow_filter_dict
+            )
+        else:
+            return self.boto_client.count_closed_workflow_executions(
+                domain=self.workflow_client_config.domain,
+                closeTimeFilter=time_filter_dict[time_filter_type],
+                **workflow_filter_dict
+            )
+
+
+def _build_time_filter_dict(oldest_start_date=None, latest_start_date=None, oldest_close_date=None, latest_close_date=None):
```

Comment:
Hmm I think it possible to combine at the filter type in this count_executions. In that case my question is do I have to document the usage in great detail so that the user won't get confused when using count_executions? Or it is up to the user to learn by having exceptions? Like to know which filters can be used together while others cannot? 

The reason that I separate count open workflow from count close workflow is their filter parameters don't really mirror each other(open workflow only take start date, while close workflow can take start & close date, and it can take a dedicated close_status filter). I could combine start & close date count APIs in count_close_workflows to reduce API numbers, how do you think of that?